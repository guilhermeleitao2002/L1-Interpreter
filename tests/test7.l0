let factorial = fn n => {
  let result = box 1;
  let i = box n;
  
  while (!i > 0) {
    result := !result * !i;
    i := !i - 1
  };
  
  !result
};

let n = 5;
println "Factorial of";
println n;
println "is:";
println ((factorial) (n));

let sum = fn n => {
  let result = box 0;
  let i = box 1;
  
  while (!i <= n) {
    result := !result + !i;
    i := !i + 1
  };
  
  !result
};

println "Sum from 1 to";
println n;
println "is:";
println ((sum) (n));;