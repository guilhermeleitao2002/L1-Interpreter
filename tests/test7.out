Factorial of
5
is:
120
Sum from 1 to
5
is:
15